pkg/nodeconfig/nodeconfig.go

```diff
@@ -120,12 +121,12 @@ func discoverKubeAPIServerEndpoint() (string, error) {
 
 // NewNodeConfig creates a new instance of nodeConfig to be used by the caller.
 // hostName having a value will result in the VM's hostname being changed to the given value.
-func NewNodeConfig(clientset *kubernetes.Clientset, clusterServiceCIDR, vxlanPort string,
+func NewNodeConfig(c client.Client, clientset *kubernetes.Clientset, clusterServiceCIDR, vxlanPort string,
```

Comment:
This is being introduced and not used in this commit

Reply:
Being used now